### PR TITLE
DAOS-3711 test: Update container ACL

### PR DESCRIPTION
**Pull request by amjustin13**
Test covers the following:
SRS-10: Container Management - container ACL allows add/remove access control - https://jira.hpdd.intel.com/browse/SRS-10

- Test that container update command performs as expected with invalid inputs in command line for the --entry and --acl-file options
- Test that container update command performs as expected with invalid inputs within ACL file provided.
- Test that container update command performs as expected with valid ACL file provided. i.e. that new file entries leave other entries undisturbed, updating already existing principals modified successfully the right entries and adding entries with new principals.
- Test that container update command fails with no permission -1001 when user doesn't have the right permissions.

Signed-off-by: Justiniano-pagn amanda.justiniano-pagn@intel.com

### CODE REVIEW

File: src/tests/ftest/util/daos_utils.py [hunk 1]
```diff
@@ -209,6 +209,28 @@ def container_delete_acl(self, pool, svc, cont, principal):
...
+            cont (str): Container for which to get the ACL.
+            entry (bool, optional): Add or modify a single ACL entry
+            acl_file (str, optional): Input file containing ACL
+
+        Returns:
+            CmdResult: Object that contains exit status, stdout, and other
+                information.
+
+        Raises:
+            CommandFailure: if the daos container get-acl command fails.
```
Comment (shimizukko) [inline src/tests/ftest/util/daos_utils.py:227]:
```suggestion
            CommandFailure: if the daos container update-acl command fails.
```

File: src/tests/ftest/security/cont_update_acl.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        self.acl_filename = "test_acl_file.txt"
+        self.daos_cmd = self.get_daos_command()
+        self.prepare_pool()
+        self.add_container(self.pool)
+
+        # List of ACL entries
+        self.cont_acl = self.get_container_acl_list(
+            self.pool.uuid, self.pool.svc_ranks[0], self.container.uuid)
+
+    def error_handling(self, results, err_msg):
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_update_acl.py:51]: I see this function and acl_file_diff in cont_overwrite_acl.py too, so you might want to move them to ContSecurityTestBase.

**Review by shimizukko — DISMISSED** [review]
LGTM

File: src/tests/ftest/security/cont_update_acl.py (hunk 2 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_update_acl.py:51]: Yes, that's a good point. I'll make that change.

**Review by kjacque — COMMENTED** [review]
No complaints from my side. Looks like you need to address Makito's comments, but once you do that, I will approve.

File: src/tests/ftest/util/daos_utils.py (hunk 1 above)
Comment (amjustin13) [reply src/tests/ftest/util/daos_utils.py:248]: Done.

File: src/tests/ftest/security/cont_update_acl.py (hunk 2 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_update_acl.py]: Done

**Review by kjacque — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]
LGTM

**Comment by amjustin13** [comment]
Test Failures:
- Macsio TEST4 avocado test timeout: https://jira.hpdd.intel.com/browse/DAOS-5591
- ERROR: daos_io_server:0 *** Process 19891 received signal 11 ***: https://jira.hpdd.intel.com/browse/DAOS-5651
- DAOS degrade mode test failed on rebuild: https://jira.hpdd.intel.com/browse/DAOS-5488

**Pull request merged**